engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -38,9 +38,13 @@ class DeleteWorkflowFilesActor(rootWorkflowId: RootWorkflowId,
   val asyncIO = new AsyncIo(ioActorRef, gcsCommandBuilder)
   val callCache = new CallCache(EngineServicesStore.engineDatabaseInterface)
 
+
   startWith(Pending, NoData)
 
   when(Pending) {
+    case Event(ioReceivable, _) if ioReceive.isDefinedAt(ioReceivable) =>
+      ioReceive.apply(ioReceivable)
+      stay
```

Comment:
Can you please explain what does this do? I am not really sure (/have forgotten) how the this helps with back-pressuring

Reply:
The `IoClientHelper` requires you to put `ioReceive orElse { ... }` in your actor code to apply its safety logic around using the IO actor (it adds timeouts and backpressure retries automatically for us). Because we're in a FSM here we don't have a single `receive` method to apply that to, so we have to add cases in every state.

You can have a look at [IoClientHelper](https://github.com/broadinstitute/cromwell/blob/develop/core/src/main/scala/cromwell/core/io/IoClientHelper.scala#L30) and then [RobustClientHelper](https://github.com/broadinstitute/cromwell/blob/develop/core/src/main/scala/cromwell/core/actor/RobustClientHelper.scala#L27) to see the extra cases which this is providing.